src/researches/metaDescriptionKeyword.js

```diff
@@ -1,19 +1,43 @@
-import matchTextWithWord from "../stringProcessing/matchTextWithWord.js";
-
-import { escapeRegExp } from "lodash-es";
+import matchTextWithArray from "../stringProcessing/matchTextWithArray";
+import getSentences from "../stringProcessing/getSentences";
+import escapeRegExp from "lodash-es/escapeRegExp";
 
 /**
  * Matches the keyword in the description if a description and keyword are available.
- * default is -1 if no description and/or keyword is specified
+ * Returns null if no description is specified in the given paper.
  *
  * @param {Paper} paper The paper object containing the description.
- * @returns {number} The number of matches with the keyword
+ * @param {Researcher} researcher the researcher object to gather researchers from.
+ * @returns { Object | null } The number of matches per keyword term, for the entire description and for each individual sentence.
  */
-export default function( paper ) {
+export default function( paper, researcher ) {
 	if ( paper.getDescription() === "" ) {
-		return -1;
+		return null;
 	}
-	var keyword = escapeRegExp( paper.getKeyword() );
-	return matchTextWithWord( paper.getDescription(), keyword, paper.getLocale() ).count;
+	const description = paper.getDescription();
+	const locale = paper.getLocale();
+
+	let keyPhraseForms = researcher.getResearch( "morphology" ).keyphraseForms;
+	keyPhraseForms = keyPhraseForms.map( keyWordForms => keyWordForms.map( form => escapeRegExp( form ) )  );
+
+	const sentences = getSentences( description );
+
+	let matchResults = { };
+
+	/*
+		Compute the number of matches for each keyword in the key phrase (morphology included).
+	   	Result is an array of match counts, e.g. [0,4,1] means keywords 2 and 3 and their
+	   	morphological forms have been matched 4 and 1 time(s) respectively.
+	 */
+	matchResults.fullDescription = keyPhraseForms.map(
+		keywordForms => matchTextWithArray( description, keywordForms, locale ).count
+	);
```

Comment:
I think it is totally valid what you are doing here, but for the sake of consistency I would invite you to consider using a function from `src/researches/findKeywordFormsInString.js`. It would improve readability of your code.

Reply:
Check out this PR for a similar sentence/text issue https://github.com/Yoast/YoastSEO.js/pull/1645.